File: android/src/main/java/com/pichillilorenzo/flutter_inappwebview/InAppBrowserOptions.java
package com.pichillilorenzo.flutter_inappwebview;

public class InAppBrowserOptions extends Options {

    public static final String LOG_TAG = "InAppBrowserOptions";

    public boolean hidden = false;
    public boolean toolbarTop = true;
    public String toolbarTopBackgroundColor = "";
    public String toolbarTopFixedTitle = "";
    public boolean hideUrlBar = false;

    public boolean hideTitleBar = false;
    public boolean closeOnCannotGoBack = true;
    public boolean progressBar = true;

    public boolean hideMenu = false;
    public boolean showToolbarBackButton = false;

    @Override
    public void setValueForField(String field, Object value) {
        switch (field) {
            case "hidden":
                if (value != null) hidden = (boolean) value;
                break;
            case "toolbarTop":
                if (value != null) toolbarTop = (boolean) value;
                break;
            case "toolbarTopBackgroundColor":
                toolbarTopBackgroundColor = value != null ? value.toString() : "";
                break;
            case "toolbarTopFixedTitle":
                toolbarTopFixedTitle = value != null ? value.toString() : "";
                break;
            case "hideUrlBar":
                if (value != null) hideUrlBar = (boolean) value;
                break;
            case "hideTitleBar":
                if (value != null) hideTitleBar = (boolean) value;
                break;
            case "closeOnCannotGoBack":
                if (value != null) closeOnCannotGoBack = (boolean) value;
                break;
            case "progressBar":
                if (value != null) progressBar = (boolean) value;
                break;
            case "hideMenu":
                if (value != null) hideMenu = (boolean) value;
                break;
            case "showToolbarBackButton":
                if (value != null) showToolbarBackButton = (boolean) value;
                break;
        }
    }

    @Override
    public Object getValueForField(String field) {
        switch (field) {
            case "hidden":
                return hidden;
            case "toolbarTop":
                return toolbarTop;
            case "toolbarTopBackgroundColor":
                return toolbarTopBackgroundColor;
            case "toolbarTopFixedTitle":
                return toolbarTopFixedTitle;
            case "hideUrlBar":
                return hideUrlBar;
            case "hideTitleBar":
                return hideTitleBar;
            case "closeOnCannotGoBack":
                return closeOnCannotGoBack;
            case "progressBar":
                return progressBar;
            case "hideMenu":
                return hideMenu;
            case "showToolbarBackButton":
                return showToolbarBackButton;
            default:
                return null;
        }
    }

    @Override
    public String[] getFieldList() {
        return new String[]{
                "hidden",
                "toolbarTop",
                "toolbarTopBackgroundColor",
                "toolbarTopFixedTitle",
                "hideUrlBar",
                "hideTitleBar",
                "closeOnCannotGoBack",
                "progressBar",
                "hideMenu",
                "showToolbarBackButton"
        };
    }
}

File: ios/Classes/InAppBrowserOptions.swift
//
//  InAppBrowserOptions.swift
//  flutter_inappwebview
//
//  Created by Lorenzo on 17/09/18.
//

import Foundation

@objcMembers
public class InAppBrowserOptions: Options {
    
    var hidden = false
    var toolbarTop = true
    var toolbarTopBackgroundColor = ""
    var toolbarTopFixedTitle = ""
    var hideUrlBar = false
    var toolbarBottom = true
    var toolbarBottomBackgroundColor = ""
    var toolbarBottomTranslucent = true
    var closeButtonCaption = ""
    var closeButtonColor = ""
    var presentationStyle = 0 //fullscreen
    var transitionStyle = 0 //crossDissolve
    var spinner = true
    
    override init(){
        super.init()
    }

    override func parse(options: [String: Any]) -> Options {
        for optionKey in options.keys {
            switch(optionKey) {
                case "hidden":
                    hidden = options[optionKey] as? Bool ?? hidden
                case "toolbarTop":
                    toolbarTop = options[optionKey] as? Bool ?? toolbarTop
                case "toolbarTopBackgroundColor":
                    toolbarTopBackgroundColor = options[optionKey] as? String ?? toolbarTopBackgroundColor
                case "toolbarTopFixedTitle":
                    toolbarTopFixedTitle = options[optionKey] as? String ?? toolbarTopFixedTitle
                case "hideUrlBar":
                    hideUrlBar = options[optionKey] as? Bool ?? hideUrlBar
                case "toolbarBottom":
                    toolbarBottom = options[optionKey] as? Bool ?? toolbarBottom
                case "toolbarBottomBackgroundColor":
                    toolbarBottomBackgroundColor = options[optionKey] as? String ?? toolbarBottomBackgroundColor
                case "toolbarBottomTranslucent":
                    toolbarBottomTranslucent = options[optionKey] as? Bool ?? toolbarBottomTranslucent
                case "closeButtonCaption":
                    closeButtonCaption = options[optionKey] as? String ?? closeButtonColor
                case "closeButtonColor":
                    closeButtonColor = options[optionKey] as? String ?? closeButtonColor
                case "presentationStyle":
                    presentationStyle = options[optionKey] as? Int ?? presentationStyle
                case "transitionStyle":
                    transitionStyle = options[optionKey] as? Int ?? transitionStyle
                case "spinner":
                    spinner = options[optionKey] as? Bool ?? spinner
            default:
                hidden = options["hidden"] as? Bool ?? hidden
            }
        }

        return self
    }

    override func getHashMap() -> [String: Any] {
        var options: [String: Any] = [:]
        
        options["hidden"] = hidden
        options["toolbarTop"] = toolbarTop
        options["toolbarTopBackgroundColor"] = toolbarTopBackgroundColor
        options["toolbarTopFixedTitle"] = toolbarTopFixedTitle
        options["hideUrlBar"] = hideUrlBar
        options["toolbarBottom"] = toolbarBottom
        options["toolbarBottomBackgroundColor"] = toolbarBottomBackgroundColor
        options["toolbarBottomTranslucent"] = toolbarBottomTranslucent
        options["closeButtonCaption"] = closeButtonCaption
        options["closeButtonColor"] = closeButtonColor
        options["presentationStyle"] = presentationStyle
        options["transitionStyle"] = transitionStyle
        options["spinner"] = spinner

        return options
    }
    
}
